src/supervisor3.erl

```diff
@@ -573,6 +578,8 @@ handle_call({delete_child, Name}, _From, State) ->
 	    {reply, ok, NState};
 	{value, #child{pid=?restarting(_)}} ->
 	    {reply, {error, restarting}, State};
+	{value, #child{pid=?delayed_restart(_)}} ->
```

Comment:
cancel timer here.

Reply:
no. sorry.

Reply:
but if the purpose of this change is to be able to delete children who are pending on delayed delete. returning error here isn't quite helpful or?

Reply:
ok, one will have to terminate it first before delete.